### ISSUE TITLE
在windows中图像旋转并保存后，使用paddleocr.PaddleOCR调用det模型检测框变少，并且和直接调用predict_det不一致

### ISSUE THREAD

**Issue by Strepsiades** [PaddlePaddle/PaddleOCR#13456]
#### 问题描述 / Problem Description

这个问题困扰了我一天：我单独用predict_det.py检测图片，检测框很正常。但是使用paddle.PaddleOCR检测一张测试图片，就会少很多检测框。

![Uploading 2c87c15e2763333574e3fb6a64c4728.jpg…]()
参考下述测试代码 img的形状不随系统中图像旋转而旋转 而img2的形状随图像旋转而旋转。

从脚本调用使用的时img1的方法，从命令行调用使用的是img2的方法。不同输入形状会导致predict_det脚本表现不同。这导致一些非常诡异的检测框丢失现象。

具体来说，如果图像是真实方向，那么检测框就会更多更全。但是如果一个图片本来是放倒的，靠调整变成正的，那么命令行调用predict_det，检测框就正常，脚本paddle.PaddleOcr()调用，就会少检测框。
#### 运行环境 / Runtime Environment
- OS:
- Paddle:
- PaddleOCR:

#### 复现代码 / Reproduction Code
```python
import numpy as np
import cv2
file_path = r"D:\AIdata\training_project\handwrite_ppocrv4\data\20240722204811.png"

# from paddleocr.py line 513
def img_decode(content: bytes):
    np_arr = np.frombuffer(content, dtype=np.uint8)
    return cv2.imdecode(np_arr, cv2.IMREAD_UNCHANGED)

with open(file_path, 'rb') as f:
    img_str = f.read()
    img = img_decode(img_str)
print(img.shape)

# from tools\infer\predict_rec.py line 655
img2 = cv2.imread(file_path)
print(img2.shape)
```

#### 完整报错 / Complete Error Message

(paddle_nlp) PS D:\AIdata\training_project\handwrite_ppocrv4>  d:; cd 'd:\AIdata\training_project\handwrite_ppocrv4'; & 'd:\miniconda\envs\paddle_nlp\python.exe' 'c:\Users\zhaoy\.vscode\extensions\ms-python.debugpy-2024.8.0-win32-x64\bundled\libs\debugpy\adapter/../..\debugpy\launcher' '54864' '--' 'd:\AIdata\training_project\handwrite_ppocrv4\test_opencv.py' 
(1045, 751, 3)
(751, 1045, 3)
(paddle_nlp) PS D:\AIdata\training_project\handwrite_ppocrv4>  d:; cd 'd:\AIdata\training_project\handwrite_ppocrv4'; & 'd:\miniconda\envs\paddle_nlp\python.exe' 'c:\Users\zhaoy\.vscode\extensions\ms-python.debugpy-2024.8.0-win32-x64\bundled\libs\debugpy\adapter/../..\debugpy\launcher' '54883' '--' 'd:\AIdata\training_project\handwrite_ppocrv4\test_opencv.py' 
(1045, 751, 3)
(1045, 751, 3)
(paddle_nlp) PS D:\AIdata\training_project\handwrite_ppocrv4>  d:; cd 'd:\AIdata\training_project\handwrite_ppocrv4'; & 'd:\miniconda\envs\paddle_nlp\python.exe' 'c:\Users\zhaoy\.vscode\extensions\ms-python.debugpy-2024.8.0-win32-x64\bundled\libs\debugpy\adapter/../..\debugpy\launcher' '54898' '--' 'd:\AIdata\training_project\handwrite_ppocrv4\test_opencv.py' 
(1045, 751, 3)
(751, 1045, 3)
#### 可能解决方案 / Possible solutions

把paddleocr.py中的img_decode改成普通的cv2.imread()

**Comment by jingsongliujing** [comment]
只是可视化结果不对吗？可以打印出文本和坐标结果看看完整不？

**Comment by Strepsiades** [comment]
识别的框也变少了，不仅是可视化的问题。比如我把这个网页横着截图，然后在windows中旋转改为正确角度。下方结果1就不能随着系统旋转而旋转，还是横着的。结果2就是正确的方向。结果少识别了OS:三个字。我自己的例子图片方向不对相比于正确可能是15个和20个框的差距。针对文字的方向修正是没用的，因为识别框本身就少了。

![result1](https://github.com/user-attachments/assets/0d6a286c-881b-4be0-bfcc-5024ae0a3170)
![result2](https://github.com/user-attachments/assets/7da9c0ad-6b27-4494-9a64-d2987a812246)

```import cv2
from paddleocr import PaddleOCR, draw_ocr

# 初始化OCR模型
ocr = PaddleOCR(use_angle_cls=True, lang='ch', det=True)  # 可根据需要选择语言

# 读取输入图片
image_path = r'D:\AIdata\training_project\handwrite_ppocrv4\data\20240723193639.png'


# 直接调研脚本进行文本检测
result1 = ocr.ocr(image_path, cls=True)
print(result1)

# 先用opencv读取
image = cv2.imread(image_path)
result2 = ocr.ocr(image, cls=True)
print(result2)

# 显示结果1
from PIL import Image 
import numpy as np

def img_decode(content: bytes):
    np_arr = np.frombuffer(content, dtype=np.uint8)
    return cv2.imdecode(np_arr, cv2.IMREAD_UNCHANGED)

with open(image_path, 'rb') as f:
    img_str = f.read()
    img = img_decode(img_str)


result = result1[0]
boxes = [line[0] for line in result]
txts = [line[1][0] for line in result]
scores = [line[1][1] for line in result]
im_show = draw_ocr(img, boxes, txts, scores)
im_show = Image.fromarray(im_show)
im_show.save('result1.jpg')

# 显示结果2
result = result2[0]
boxes = [line[0] for line in result]
txts = [line[1][0] for line in result]
scores = [line[1][1] for line in result]
im_show = draw_ocr(image, boxes, txts, scores)
im_show = Image.fromarray(im_show)
im_show.save('result2.jpg')```

**Comment by UserWangZz** [comment]
感觉是不是图像读取的问题，可以看看检测前输入模型的图像shape是不是一致的？

**Comment by Strepsiades** [comment]
> 感觉是不是图像读取的问题，可以看看检测前输入模型的图像shape是不是一致的？

是的。目前只要图片旋转过，通过predict_rec.py和通过paddleocr.PaddleOCR调用时，检测前图片shape就不一样，检测后结果也不一样。如果文档是横着的（但是系统里看着是正着的）可能会少一些框。

**Comment by UserWangZz** [comment]
https://blog.csdn.net/sxrkm6s4csx/article/details/117525646
参考一下这个文章，是不是图片里的方向信息导致你读图出现了问题
